argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,174 @@
+# <Copyright 2020, Argo AI, LLC. Released under the MIT license.>
+"""Detection utilities for the Argoverse detection leaderboard.
+
+Accepts detections (in Argoverse ground truth format) and ground truth labels
+for computing evaluation metrics for 3d object detection. We have five different,
+metrics: mAP, ATE, ASE, AOE, and DCS. A true positive for mAP is defined as the
+highest confidence prediction within a specified euclidean distance threshold
+from a bird's-eye view. We prefer these metrics instead of IoU due to the
+increased interpretability of the error modes in a set of detections.
+
+"""
+
+from enum import Enum, auto
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from scipy.spatial.transform import Rotation as R
+
+from argoverse.utils.transform import quat_first2last
+
+
+class SimFnType(Enum):
+    CENTER = auto()
+
+
+class DistFnType(Enum):
+    TRANSLATION = auto()
+    SCALE = auto()
+    ORIENTATION = auto()
+
+
+class InterpType(Enum):
+    ALL = auto()
+
+
+class FilterMetric(Enum):
+    EUCLIDEAN = auto()
+
+
+def filter_instances(
+    instances: np.ndarray,
+    target_class: str,
+    filter_metric: FilterMetric = FilterMetric.EUCLIDEAN,
+    max_dist: float = 50.0,
+) -> np.ndarray:
+    """Filter the annotations based on a set of conditions.
+
+    Args:
+        annos: The instances to be filtered.
+        target_class: The name of the class of interest.
+        filter_metric: The range metric used for filtering.
+
+    Returns:
+        The filtered annotations.
+    """
+    instances = np.array([instance for instance in instances if instance.label_class == target_class])
+
+    if filter_metric == FilterMetric.EUCLIDEAN:
+        centers = np.array([dt.translation for dt in instances])
+        filtered_annos = np.array([])
+
+        if centers.shape[0] > 0:
+            dt_dists = np.linalg.norm(centers, axis=1)
+            filtered_annos = instances[dt_dists < max_dist]
+
+        return filtered_annos
+
+
+def get_ranks(dts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
not sure that the input type here is correct if we can call `dt.score` on it. Should it be `List[ObjectLabelRecord])`?